app/web_ui/src/routes/(app)/docs/library/[project_id]/upload_file_dialog.svelte

```diff
@@ -329,51 +357,98 @@
-          on:dragleave={handleDragLeave}
-          on:drop={handleDrop}
-          on:click={openFileDialog}
-          role="button"
-          tabindex="0"
-          on:keydown={(e) => {
-            if (e.key === "Enter" || e.key === " ") {
-              e.preventDefault()
-              openFileDialog()
-            }
-          }}
-        >
-          <div class="space-y-2">
-            <div class="w-10 h-10 mx-auto text-gray-500">
-              <UploadIcon />
-            </div>
-            <div>
-              <p class="text-gray-500">Drop files here or click to select</p>
+        <div class="pb-2">
+          <!-- Dropzone -->
+          <div
+            class="border-2 border-dashed rounded-lg p-8 text-center transition-colors cursor-pointer {drag_over
+              ? 'border-primary bg-primary/5'
+              : 'border-gray-300 hover:border-gray-400'}"
+            on:dragover={handleDragOver}
+            on:dragleave={handleDragLeave}
+            on:drop={handleDrop}
+            on:click={openFileDialog}
+            role="button"
+            tabindex="0"
+            on:keydown={(e) => {
+              if (e.key === "Enter" || e.key === " ") {
+                e.preventDefault()
+                openFileDialog()
+              }
+            }}
+          >
+            <div class="space-y-2">
+              <div class="w-10 h-10 mx-auto text-gray-500">
+                <UploadIcon />
+              </div>
+              <div>
+                <p class="text-gray-500">Drop files here or click to select</p>
+              </div>
             </div>
           </div>
+
+          <!-- Hidden file input -->
+          <input
+            bind:this={file_input}
+            type="file"
+            multiple
+            class="hidden"
+            on:change={handleFileSelect}
+            accept={supported_file_types.join(",")}
+          />
+
+          {#if unsupported_files_count > 0}
+            <div class="text-error text-sm">
+              {unsupported_files_count} file{unsupported_files_count === 1
+                ? ""
+                : "s"} skipped due to unsupported format
+            </div>
+          {/if}
         </div>
 
-        <!-- Hidden file input -->
-        <input
-          bind:this={file_input}
-          type="file"
-          multiple
-          class="hidden"
-          on:change={handleFileSelect}
-          accept={supported_file_types.join(",")}
-        />
-
-        {#if unsupported_files_count > 0}
-          <div class="text-error text-sm">
-            {unsupported_files_count} file{unsupported_files_count === 1
-              ? ""
-              : "s"} skipped due to unsupported format
+        <!-- Tag selection -->
+        <div>
+          <FormElement
+            inputType="header_only"
+            label="Tags"
+            id="tags_section"
+            description="Add tags to organize your documents"
+            info_description="Add tags to organize your documents"
```

Comment:
"Any tags set here will be added to each document you add. Tags can be used to filter your document set."